src/analysis/special_functions/trigonometric.lean

```diff
@@ -1771,55 +1772,48 @@ end
 open filter
 open_locale topological_space
 
-lemma tendsto_sin_pi_div_two : tendsto sin (𝓝[set.Iio (π/2)] (π/2)) (𝓝 1) :=
-by { convert (@continuous.continuous_within_at _ _ _ _ _ (set.Iio (π/2)) (π/2) continuous_sin).tendsto,
-  simp }
+lemma tendsto_sin_pi_div_two : tendsto sin (𝓝[Iio (π/2)] (π/2)) (𝓝 1) :=
+begin
+  rw show (𝓝 1 = 𝓝 (sin (π/2))), by rw sin_pi_div_two,
+  exact continuous_sin.continuous_within_at
+end
 
-lemma tendsto_cos_pi_div_two : tendsto cos (𝓝[set.Iio (π/2)] (π/2)) (𝓝[set.Ioi 0] 0) :=
+lemma tendsto_cos_pi_div_two : tendsto cos (𝓝[Iio (π/2)] (π/2)) (𝓝[Ioi 0] 0) :=
 begin
-  change (tendsto cos (𝓝[set.Iio (π/2)] (π/2)) (_ ⊓ _)),
-  rw tendsto_inf,
-  split,
-  { convert (@continuous.continuous_within_at _ _ _ _ _ (set.Iio (π/2)) (π/2) continuous_cos).tendsto,
-    simp },
-  { rw tendsto_principal,
-    use set.Ioi (0:ℝ),
-    split,
-    { exact mem_nhds_sets is_open_Ioi (by simp [pi_div_two_pos]) },
-    { use set.Iio (π/2),
-      split,
-      { exact λ (x:ℝ), set.mem_Iio.mpr },
-      { intro x,
-        simp,
-        intros h1 h2,
-        exact cos_pos_of_mem_Ioo (by linarith) h2 } } },
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within,
+  rw show (𝓝 0 = 𝓝 (cos (π/2))), by rw cos_pi_div_two,
+  exact continuous_cos.continuous_within_at,
+  have : Ioo (-(π / 2)) (π / 2) ∈ 𝓝[Iio (π / 2)] (π / 2) := Ioo_mem_nhds_within_Iio
+    (right_mem_Ioc.mpr $ by linarith [pi_div_two_pos, neg_neg_of_pos pi_div_two_pos]),
+  filter_upwards [this],
+  exact λ x hx, cos_pos_of_mem_Ioo hx.1 hx.2
 end
 
-lemma tendsto_tan_pi_div_two : tendsto tan (𝓝[set.Iio (π/2)] (π/2)) at_top :=
+lemma tendsto_tan_pi_div_two : tendsto tan (𝓝[Iio (π/2)] (π/2)) at_top :=
 begin
   convert tendsto_mul_at_top (by norm_num) (tendsto.inv_tendsto_zero tendsto_cos_pi_div_two)
             tendsto_sin_pi_div_two,
   ext x,
-  ring,
-  apply tan_eq_sin_div_cos x,
+  rw tan_eq_sin_div_cos x,
+  ring
```

Comment:
This is a tiny change but it removes a non-terminal `ring`. This tactic has several normal forms it can use, see https://leanprover-community.github.io/mathlib_docs/tactics.html#ring. If the default normal form changes one day then your proof may break.